sympy/tensor/array/nditer.py

```diff
@@ -0,0 +1,58 @@
+from sympy.tensor.array.dense_ndim_array import DenseNDimArray
+from sympy.tensor.array.sparse_ndim_array import SparseNDimArray
+from sympy.tensor.array.ndim_array import NDimArray
+from sympy.core.compatibility import Iterable
+
+class nditer(object):
```

Comment:
SymPy standard is `NDIter`

Reply:
This should be a SymPy object, so inherit `Expr`.

Reply:
Maybe this object should be called `Flatten`, as what it does it flattening the array.